crates/red_knot_python_semantic/src/semantic_index/builder.rs

```diff
@@ -426,6 +412,20 @@ where
                         builder.pop_scope()
                     },
                 );
+
+                let symbol = self
+                    .add_or_update_symbol(function_def.name.id.clone(), SymbolFlags::IS_DEFINED);
+                self.add_definition(symbol, function_def);
+
+                // The default value of the parameters needs to be evaluated in the
+                // enclosing scope.
+                for default in function_def
+                    .parameters
+                    .iter_non_variadic_params()
+                    .filter_map(|param| param.default.as_deref())
+                {
+                    self.visit_expr(default);
+                }
```

Comment:
While we're fixing the order of things here, I think this also belongs above the definition of the function name:

```
>>> def bool(x = bool):
...     return x
...
>>> bool()
<class 'bool'>
```

Would you be open to also adding a test for this and fixing the ordering?

Reply:
Done! Added another "no panic" test and then a mostly copy-pasted unit test that doesn't really check for much but is a baby step towards the sort of thing you mentioned in your other comment.

Reply:
Thank you! As I mentioned in my other comment, I think the new copy-pasted test doesn't add enough value to be worth adding right now, so we can remove it. The inference test we eventually want would be at the type-inference level, not the semantic level. Everything else looks great!